src/webui/api/torrentscontroller.cpp

```diff
@@ -422,61 +422,58 @@ void TorrentsController::propertiesAction()
-    dataDict[KEY_PROP_ADDITION_DATE] = toTimeStamp(torrent->addedTime());
-    if (torrent->hasMetadata())
-    {
-        dataDict[KEY_PROP_LAST_SEEN] = toTimeStamp(torrent->lastSeenComplete());
-        dataDict[KEY_PROP_COMPLETION_DATE] = toTimeStamp(torrent->completedTime());
-        dataDict[KEY_PROP_CREATION_DATE] = toTimeStamp(torrent->creationDate());
-    }
-    else
+    const bool hasMetadata = torrent->hasMetadata();
+
+    const QJsonObject ret
     {
-        dataDict[KEY_PROP_LAST_SEEN] = -1;
-        dataDict[KEY_PROP_COMPLETION_DATE] = -1;
-        dataDict[KEY_PROP_CREATION_DATE] = -1;
-    }
-    dataDict[KEY_PROP_SAVE_PATH] = torrent->savePath().toString();
-    dataDict[KEY_PROP_DOWNLOAD_PATH] = torrent->downloadPath().toString();
-    dataDict[KEY_PROP_COMMENT] = torrent->comment();
+        {KEY_TORRENT_INFOHASHV1, infoHash.v1().toString()},
+        {KEY_TORRENT_INFOHASHV2, infoHash.v2().toString()},
+        {KEY_TORRENT_NAME, torrent->name()},
+        {KEY_TORRENT_ID, torrent->id().toString()},
+        {KEY_PROP_TIME_ELAPSED, torrent->activeTime()},
+        {KEY_PROP_SEEDING_TIME, torrent->finishedTime()},
+        {KEY_PROP_ETA, torrent->eta()},
+        {KEY_PROP_CONNECT_COUNT, torrent->connectionsCount()},
+        {KEY_PROP_CONNECT_COUNT_LIMIT, torrent->connectionsLimit()},
+        {KEY_PROP_DOWNLOADED, torrent->totalDownload()},
+        {KEY_PROP_DOWNLOADED_SESSION, torrent->totalPayloadDownload()},
+        {KEY_PROP_UPLOADED, torrent->totalUpload()},
+        {KEY_PROP_UPLOADED_SESSION, torrent->totalPayloadUpload()},
+        {KEY_PROP_DL_SPEED, torrent->downloadPayloadRate()},
+        {KEY_PROP_DL_SPEED_AVG, (torrent->totalDownload() / ((dlDuration == 0) ? -1 : dlDuration))},
+        {KEY_PROP_UP_SPEED, torrent->uploadPayloadRate()},
+        {KEY_PROP_UP_SPEED_AVG, (torrent->totalUpload() / ((ulDuration == 0) ? -1 : ulDuration))},
+        {KEY_PROP_DL_LIMIT, ((downloadLimit <= 0) ? -1 : downloadLimit)},
+        {KEY_PROP_UP_LIMIT, ((uploadLimit <= 0) ? -1 : uploadLimit)},
+        {KEY_PROP_WASTED, torrent->wastedSize()},
+        {KEY_PROP_SEEDS, torrent->seedsCount()},
+        {KEY_PROP_SEEDS_TOTAL, torrent->totalSeedsCount()},
+        {KEY_PROP_PEERS, torrent->leechsCount()},
+        {KEY_PROP_PEERS_TOTAL, torrent->totalLeechersCount()},
+        {KEY_PROP_RATIO, ((ratio > BitTorrent::Torrent::MAX_RATIO) ? -1 : ratio)},
+        {KEY_PROP_REANNOUNCE, torrent->nextAnnounce()},
+        {KEY_PROP_TOTAL_SIZE, torrent->totalSize()},
+        {KEY_PROP_PIECES_NUM, torrent->piecesCount()},
+        {KEY_PROP_PIECE_SIZE, torrent->pieceLength()},
+        {KEY_PROP_PIECES_HAVE, torrent->piecesHave()},
+        {KEY_PROP_CREATED_BY, torrent->creator()},
+        {KEY_PROP_ISPRIVATE, torrent->isPrivate()},
+        {KEY_PROP_ADDITION_DATE, toTimeStamp(torrent->addedTime())},
+        {KEY_PROP_LAST_SEEN, (hasMetadata ? toTimeStamp(torrent->lastSeenComplete()) : -1)},
```

Comment:
Are all such checks for hasMetadata really needed?

Reply:
Not sure. I didn't investigate that deep since it wasn't related to the topic.

Reply:
I removed them since it seems to work and doesn't crash.